### DISCUSSION TITLE
Cant get --log to change log loaction

### DISCUSSION THREAD

**Discussion by DanHSO** [microsoft/winget-pkgs#86162] [Q&A]
So i am developing a solution to distribute Apps on demand in our Org using Winget.
I am close to having it work but sometimes i need logs to spot issues with either local issues or firewall etc.. Now the problem is that i am using Ivanti Automation to actually do the scripts and it runs by default inn system context.

And here lies what i am thinking is the root of the problem. %localappdata%\ etc etc dosent really exsist so the logs are never generated at the default location.

So for example i use winget.exe Install -e -h --accept-source-agreements --accept-package-agreements --id Notepad++.Notepad++ --log "C:\Windows\Temp\Winget\Notepad.txt"

Nothing ever gets made in said location. Doesnt mather if i pre-create the folder and file. Am i missing something fundamental here? Or is the --log not functioning correctly.

**Comment by SpecterShell** [comment]
The log path passed to WinGet will be passed to the installers by appending the parameters while executing the installers.
For example if you pass the log path to WinGet like this:
```CMD
winget install Microsoft.VCRedist.2010.x64 --log "C:\the.log"
```
It will be passed to the installer like this:
```
"C:\Users\<Username>\AppData\Local\Temp\WinGet\Microsoft.VCRedist.2010.x64.10.0.40219\vcredist_x64.exe" /passive /install /norestart /log "C:\the.log"
```

However, not all the installers support changing the logging directory, especially those built by NSIS (Nullsoft Scriptable Install System) whose logging function and parameter parsing implementation are also scriptable and optional, so WinGet doesn't know which parameter should be used for them and thus pass nothing. Usually their logs will be generated in `%temp%`, `%appdata%\<AppName>` or `%localappdata%\<AppName>`.
